docs/book/component-guide/model-registries/vertex.md

```diff
@@ -0,0 +1,150 @@
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+@step(enable_cache=False)
+def model_register() -> Annotated[str, ArtifactConfig(name="model_registry_uri")]:
+    """Model registration step."""
+    # Get the current model from the context
+    current_model = get_step_context().model
+
+    client = Client()
+    model_registry = client.active_stack.model_registry
+    model_version = model_registry.register_model_version(
+        name=current_model.name,
+        version=str(current_model.version),
+        model_source_uri=current_model.get_model_artifact("sklearn_classifier").uri,
+        description="ZenML model registered after promotion",
+    )
+    logger.info(
+        f"Model version {model_version.version} registered in Model Registry"
+    )
+    
+    return model_version.model_source_uri
+```
+
+### Configuration Options
+
+The Vertex AI Model Registry accepts the following configuration options:
+
+* `location`: The GCP region where the model registry will be created (e.g., "us-central1")
+* `project_id`: (Optional) The GCP project ID. If not specified, will use the default project
+* `credentials`: (Optional) GCP credentials configuration
+
+### Working with Model Versions
+
+Since the Vertex AI Model Registry only supports version-level operations, here's how to work with model versions:
+
+```shell
+# List all model versions
+zenml model-registry models list-versions <model-name>
+
+# Get details of a specific model version
+zenml model-registry models get-version <model-name> -v <version>
+
+# Delete a model version
+zenml model-registry models delete-version <model-name> -v <version>
+```
+
+### Key Differences from MLflow Model Registry
+
+Unlike the MLflow Model Registry, the Vertex AI implementation has some important differences:
+
+1. **Version-Only Interface**: Vertex AI only supports model version operations. You cannot register, delete, or update models directly - only their versions.
+2. **Authentication**: Uses GCP service connectors for authentication, similar to other Vertex AI services in ZenML.
+3. **Staging Levels**: Vertex AI doesn't have built-in staging levels (like Production, Staging, etc.) - these are handled through metadata.
+4. **Default Container Images**: Vertex AI requires a serving container image URI, which defaults to the scikit-learn prediction container if not specified.
+5. **Managed Service**: As a fully managed service, you don't need to worry about infrastructure management, but you need valid GCP credentials.
+
+### Limitations
+
+Based on the implementation, there are some limitations to be aware of:
+
+1. The `register_model()`, `update_model()`, and `delete_model()` methods are not implemented as Vertex AI only supports registering model versions
+2. Model stage transitions (Production, Staging, etc.) are not natively supported
+3. Models must have a serving container image URI specified or will use the default scikit-learn image
```

Comment:
how do we do that?

Reply:
updated to this which explains how vertex doesn't support creating empty models:

```
Important: The Vertex AI Model Registry implementation only supports the model version interface, not the model interface. This means you cannot register, delete, or update models directly - you can only work with model versions. Operations like `register_model()`, `delete_model()`, and `update_model()` are not supported.

Unlike platforms like MLflow where you first create a model container and then add versions to it, Vertex AI combines model registration and versioning into a single operation:

- When you upload a model, it automatically creates both the model and its first version
- Each subsequent upload with the same display name creates a new version
- You cannot create an empty model container without a version
```